src/main/java/com/microsoft/jenkins/azuread/GraphProxy.java

```diff
@@ -26,22 +26,22 @@
 import org.kohsuke.stapler.StaplerProxy;
 import org.kohsuke.stapler.StaplerRequest;
 import org.kohsuke.stapler.StaplerResponse;
-
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.concurrent.TimeUnit;
 import java.util.stream.Collectors;
 
+import static com.microsoft.jenkins.azuread.AzureSecurityRealm.addProxyToHttpClientIfRequired;
+
 /**
  * Proxies calls to the Microsoft Graph API.
  */
 @Extension
 @Restricted(NoExternalUse.class)
 public class GraphProxy implements RootAction, StaplerProxy {
-
-    private static final OkHttpClient CLIENT = new OkHttpClient();
+    private static final OkHttpClient CLIENT = addProxyToHttpClientIfRequired(new OkHttpClient().newBuilder()).build();
```

Comment:
I don't think you want this to be static, it should be created each time otherwise proxy configuration won't be updated (and possibly not loaded at this point)

Reply:
Yes, you are right! From my understanding, it should set the proxy atleast upon initially loading the plugin, right? However, in my local tests, the proxy is still being ignored.

Reply:
This is really confusing. I changed the client to a local variable and the debugger confirms that the proxy is set to "DIRECT" - yet, it still connects to the proxy. Perhaps there are other requests that also need the proxy?
If you have any ideas, let me know.

Reply:
so this is going through the proxy but it's still not working is what you're saying?

Reply:
The proxy settings seem to be applied correctly to this one request, yes.

I must apologize, I will be on vacation until Tuesday. Looking forward to fixing this afterwards. Let me know if you have ideas.

Reply:
I've had a look and I'm wondering if it's a bug in the older version of ok http we use.

I can confirm the same behaviour you are seeing =/

Reply:
Looks like we're on the latest version, maybe the logging here is helpful? https://github.com/square/okhttp/pull/5300

I've tried debugging through it in the `RouteSelector` class, one time it get's the direct proxy and the rest of the time it gets the http one, I couldn't understand why.

Reply:
Think I fixed it now